API.md

```diff
@@ -0,0 +1,66 @@
+# Octofet API
+
+## `class Octofet`
+
+Uses: class Spi2Parallel
+
+Creates an object of type `Octofet` to communicate with a particular [Octofet board](https://my.amperka.com/modules/octofet).
+
+One Octofet board includes 8 power switches. The following conventions are used: `pin` is a single power switch and `port` is a single Octofet device.
+
+### `Octofet(uint8_t pinCS, uint8_t devicesCount = 1)`
+
+Constructs a new object of the data bus hardware SPI with arguments:
+
+- `pinCS`: pin Chip Select is used to control the shift-register latch. Any GPIO-pin can be used.
+- `devicesCount`: quantity of the Octofet boards connected in a chain: the logical output pin of one module is connected to the input of another. If the `devicesCount` argument is omitted, it controls only one Octofet board.
+
+### `Octofet(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1)`
+
+Constructs a new object of the data bus software SPI with arguments:
+
+- `pinCS`: pin Chip Select is used to control the shift-register latch. Any GPIO-pin can be used.
+- `pinMOSI`: pin MOSI is used to send the data to the peripherals. Any GPIO-pin can be used.
+- `pinSCK`: pin SCK is used for the clock pulses which synchronize data transmission generated by the master. Any GPIO-pin can be used.
+- `devicesCount`: quantity of the Octofet boards connected in a chain: the logical output pin of one module is connected to the input of another. If the `devicesCount` argument is omitted, it controls only one Octofet board.
+
+The difference between the hardware and the software SPI is in transmission speed and availability. The software SPI will run much slower than the hardware SPI, but the hardware SPI works only with specific pins: `MOSI`, `MISO`(not used in Octofet) and `SCK`. Use the software SPI only when the hardware SPI pins are already used by other devices.
+
+### `void begin(const uint8_t* resetValues = NULL)`
```

Comment:
Use `nullptr` in modern C++. Do not use `NULL`: it is not type checked strictly.

```suggestion
### `void begin(const uint8_t* resetValues = nullptr)`
```